crates/sui-graphql-rpc/src/types/transaction_block_effects.rs

```diff
@@ -56,21 +66,19 @@ pub(crate) type CUnchangedSharedObject = JsonCursor<usize>;
 pub(crate) type CObjectChange = JsonCursor<usize>;
 pub(crate) type CBalanceChange = JsonCursor<usize>;
 
+type CEvent = cursor::JsonCursor<usize>;
+
 /// The effects representing the result of executing a transaction block.
 #[Object]
 impl TransactionBlockEffects {
     /// The transaction that ran to produce these effects.
     async fn transaction_block(&self) -> Result<Option<TransactionBlock>> {
-        self.tx_data
-            .as_ref()
-            .left()
-            .map(|stored_tx| TransactionBlock::try_from(stored_tx.clone()).extend())
-            .transpose()
+        Ok(Some(self.clone().try_into()?))
```

Comment:
Does this need an `extend`?

```suggestion
        Ok(Some(self.clone().try_into().extend()?))
```